controllers/xsite_config.go

```diff
@@ -112,18 +139,40 @@ func appendRemoteLocation(ctx context.Context, infinispan *ispnv1.Infinispan, re
 func appendKubernetesRemoteLocation(ctx context.Context, infinispan *ispnv1.Infinispan, remoteLocationName string, remoteKubernetes *kube.Kubernetes, logger logr.Logger, eventRec record.EventRecorder, xsite *config.XSite) error {
 	remoteNamespace := infinispan.GetRemoteSiteNamespace(remoteLocationName)
 	remoteServiceName := infinispan.GetRemoteSiteServiceName(remoteLocationName)
+	remoteRouteName := infinispan.GetRemoteSiteRouteName(remoteLocationName)
 
+	var host string
+	var port int32
+	var err error
+
+	// Try to find the service first
 	siteService := &corev1.Service{}
-	err := remoteKubernetes.Client.Get(ctx, types.NamespacedName{Name: remoteServiceName, Namespace: remoteNamespace}, siteService)
+	err = remoteKubernetes.Client.Get(ctx, types.NamespacedName{Name: remoteServiceName, Namespace: remoteNamespace}, siteService)
 	if err != nil {
 		logger.Error(err, "could not get x-site service in remote cluster", "site service name", remoteServiceName, "site namespace", remoteNamespace)
-		return err
+		if !errors.IsNotFound(err) {
+			// another error
+			return err
+		}
 	}
 
-	host, port, err := getCrossSiteServiceHostPort(siteService, remoteKubernetes, logger, eventRec, "XSiteRemoteServiceUnsupported", ctx)
-	if err != nil {
-		logger.Error(err, "error retrieving remote x-site service information")
-		return err
+	// service found
+	if err == nil {
+		host, port, err = getCrossSiteServiceHostPort(siteService, remoteKubernetes, logger, eventRec, "XSiteRemoteServiceUnsupported", ctx)
+		if err != nil {
+			logger.Error(err, "error retrieving remote x-site service information")
+			return err
+		}
+	} else {
+		// Try searching for the Route
+		siteRoute := &routev1.Route{}
+		err = remoteKubernetes.Client.Get(context.TODO(), types.NamespacedName{Name: remoteRouteName, Namespace: remoteNamespace}, siteRoute)
```

Comment:
We need to use the `ctx` provided as a parameter.

Reply:
fixed!